packages/monaco/src/typings/monaco/index.d.ts

```diff
@@ -92,53 +95,80 @@ declare module monaco.editor {
         /**
          * @internal
          */
-        // https://github.com/theia-ide/vscode/blob/standalone/0.19.x/src/vs/editor/browser/editorBrowser.ts#L644-L647
+        // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/editor/browser/editorBrowser.ts#L736
         setDecorations(decorationTypeKey: string, ranges: editorCommon.IDecorationOptions[]): void;
 
         /**
          * @internal
          */
-        // https://github.com/theia-ide/vscode/blob/standalone/0.19.x/src/vs/editor/browser/editorBrowser.ts#L654-L657
+        // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/editor/browser/editorBrowser.ts#L746
         removeDecorations(decorationTypeKey: string): void;
     }
 
-    // https://github.com/theia-ide/vscode/blob/standalone/0.20.x/src/vs/editor/browser/services/bulkEditService.ts#L14
+    // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/platform/undoRedo/common/undoRedo.ts#L92-L111
+    export class UndoRedoSource {
```

Comment:
In general, I agree that we shouldn't add unused interfaces but I also think that if we add an interface - we should add it in it's entirety even if not all props are used - as next programmer that will use it would benefit from knowing what all props are - and not "invent" something that isn't in sync with VS Code's implementation - those "inventions" later make changes like this refactoring more difficult - as you need to figure out what VS Code's implementation is and what differs it from Theia's. For instance (not a great example but nevertheless) the Theia "OnType" functionality used by quickOpen on 0.20.x isn't exactly same as in VSCode - doesn't mean it isn't as good - but it deviates from it and later could be potentially require more work to conform it to a later version.

In this example - what would you do with `undoRedoSource` ?

Reply:
I see your point of view.
I believe that it's enough to declare empty class with a reference to VS Code, I did something similar at the previous upgrade [here](https://github.com/eclipse-theia/theia/blob/47a7cfcc934faaa3943602a037d75e4752e85fee/packages/monaco/src/typings/monaco/index.d.ts#L541).

So, you have 
```
export interface IBulkEditOptions
```
with a field `undoRedoSource?: UndoRedoSource;`
and
```
   // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/platform/undoRedo/common/undoRedo.ts#L92-L111
    export class UndoRedoSource {}
```
So, everyone can see that `IBulkEditOptions` contains `undoRedoSource` field of `UndoRedoSource` type and can quickly in one click go to VS Code implementation. At the same time `index.d.ts` file doesn't contains unused class.

WDYT?

Reply:
If this is the norm then it's fine. Fixed
Thanks